### ISSUE TITLE
autofill OSX/Firefox

### ISSUE THREAD

**Issue by barajus** [subdavis/Tusk#171]
## This issue is a 

- [x] bug
- [ ] feature
- [ ] question

Hi !
I am using Firefox 60.0.1 on OSX 10.13.4.
Is autofill is working for you guys ?

No problems related to the database, but nothing is happening with TAB or click on a login field.

cheers :)

**Comment by BasixKOR** [comment]
Did you try open Tusk on the sidebar and click on the magic stick icon? It is working on my Linux machine with Firefox.

**Comment by zmilonas** [comment]
Can you provide more detail? Steps to reproduce? Do you have keyboard navigation enabled in the settings?

I cannot reproduce on the same combination of browser/os you posted.

**Comment by subdavis** [comment]
I'm going to guess @barajus is just expecting Tusk to work like LastPass or similar where the extension automatically injects options onto the page.

Trouble with that is Tusk doesn't require access to modify all pages by default, so you need to press the autofill button from inside the pop-up. 

Tusk also takes an explicit-over-implicit approach to performing all actions, such as database decryption, so it never does this without the user's input.

Automatic fill would require Tusk to either stay decrypted all the time, cache encrypted data in the clear, or perform decryption every time it sees a login box, and none of those are good options.  ChromeIPass, Kee, and others do the first option, staying decrypted all the time, but that's more acceptable because their processes are totally isolated from the browser.

I hope that helps!

If I'm wrong and you actually click the Magic wand and nothing happens, please LMK.

**Comment by barajus** [comment]
My bad, and thank you for the further details @subdavis.
I simply thought the new feature was about automatic fill.
keep up the good work :)

**Comment by jnizniowski** [comment]
I'm sorry for getting back to the old issue, but it makes me curious.
Why Tusk couldn't work in this scenario:

1. open the website with a login form
2. launch Tusk (I prefer a keyboard shortcut)
3. Fill password if needed (=decrypt database)
4. Then, if there is a matching entry and Tusk has already access to the site, the login and password are automatically filled, and Tusk is closed. If there is more than one matching entry, it stays open until you choose one of the options (well, just like it works now).

**Comment by subdavis** [comment]
Could you provide an example of something else that works like this?

I think that behavior would surprise and frustrate users.  

I'd get issues like "Im trying to open tusk but on some sites it just closes immediately".

It could stay open after the fill, but most users wouldn't notice the auto fill because they would continue to use tusk in the old pattern and search for / fill their password manually a second time

**Comment by jnizniowski** [comment]
I switched from ChromeIPass when I realized it's depreciated and not recommended. I get used to pressing Ctrl+Shift+P to autofill a form. 
I understand that my suggestion isn't perfect and might be confusing. How about a separate keyboard shortcut dedicated to autofill attempt? If the base hasn't been decrypted, it would ask for a database password. If the extension hasn't had access the page, it would ask for it. If there hasn't been any password or multiple passwords, it would show the info. 

It's a shame that, as a non-programmer, I can't help more than sharing my thoughts.
